### PR TITLE
Use widget state to manage query params

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
Widget state is used to manage query params and eslint issues are fixed.

### CODE REVIEW

File: components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/DateRangePicker.jsx [hunk 1]
```diff
@@ -18,14 +19,44 @@
...
 import React from 'react';
 import Widget from '@wso2-dashboards/widget';
-import {MuiThemeProvider, MenuItem, SelectField, FlatButton} from 'material-ui';
-import {NotificationSync, NotificationSyncDisabled} from 'material-ui/svg-icons'
-import GranularityModeSelector from './GranularityModeSelector';
+import FlatButton from 'material-ui/FlatButton';
+import MenuItem from 'material-ui/MenuItem';
+import SelectField from 'material-ui/SelectField';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+
```
Comment (Niveathika92) [inline components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/DateRangePicker.jsx]: Remove unnecessary lines

**Comment by Niveathika92** [comment]
@ruwiniwj PR merged as EI analytics was pending, please correct the commented issue in the next PR

File: components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/CustomTimeRangeSelector.jsx [hunk 2]
```diff
@@ -17,162 +18,176 @@
  */
 
 import React from 'react';
-import {MenuItem, SelectField, RaisedButton} from 'material-ui';
+import { MenuItem, SelectField, RaisedButton } from 'material-ui';
```
Comment (TanyaM) [inline components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/CustomTimeRangeSelector.jsx]: Use named imports

File: components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx [hunk 3]
```diff
@@ -158,57 +95,86 @@ export default class GranularityModeSelector extends React.Component {
...
-            <FlatButton
-                onClick={this.handleClick}
-                label='Custom'/>;
+        const {
+            getDateTimeRangeInfo, options, onChangeCustom, theme,
+        } = this.props;
+        const { open, anchorEl } = this.state;
+        const selectedTimeRange = getDateTimeRangeInfo().tr || '';
+        const customButton = selectedTimeRange === 'custom'
+            ? (
```
Comment (TanyaM) [inline components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx]: wrong formatting

File: components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/CustomTimeRangeSelector.jsx (hunk 2 above)
Comment (ruwiniwj) [reply components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/CustomTimeRangeSelector.jsx]: Fixed with PR https://github.com/wso2/analytics-solutions/pull/270

File: components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx (hunk 3 above)
Comment (ruwiniwj) [reply components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx]: This file is removed with PR https://github.com/wso2/analytics-solutions/pull/270

**Pull request merged**